src/browser/NativeMessagingBase.cpp

```diff
@@ -131,13 +131,12 @@ void NativeMessagingBase::sendReply(const QString& reply)
 
 QString NativeMessagingBase::getLocalServerPath() const
 {
-#if defined(Q_OS_WIN)
-    return QStandardPaths::writableLocation(QStandardPaths::TempLocation) + "/kpxc_server";
-#elif defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
+#if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
     // Use XDG_RUNTIME_DIR instead of /tmp/ if it's available
     QString path = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation) + "/kpxc_server";
     return path.isEmpty() ? "/tmp/kpxc_server" : path;
-#else // Q_OS_MAC and others
-    return "/tmp/kpxc_server";
+#else // Q_OS_MAC, Q_OS_WIN and others
+    QString path = QStandardPaths::writableLocation(QStandardPaths::TempLocation) + "/kpxc_server";
+    return path.isEmpty() ? "/tmp/kpxc_server" : path;
```

Comment:
The check path.isEmpty() will **never** fail due to the `+ "/kpxc_server"` one line above.
Same is true for the OS_UNIX case above, I'm actually browsing the issues because I wanted to fix that ;)

In the MacOS & others case, `writableLocation(QStandardPaths::TempLocation)` never returns an empty value (as per  [QStandardLocation docs](http://doc.qt.io/qt-5/qstandardpaths.html#StandardLocation-enum) ) - in other words: the check for `.isEmpty()` is redundant here.

In the Linux case, `writableLocation(QStandardPaths::RuntimeLocation)` **can** return an empty string, leading to `/kpxc_server` as the return value.

I can file a seperate pull request for the Linux case if that's preferred.

Reply:
Thank you! I will do the necessary fixes.